### DISCUSSION TITLE
change docker host ip address

### DISCUSSION THREAD

**Discussion by lukaszgiez** [squidfunk/mkdocs-material#4483] [Q&A] [answered]
Hi,

When I serve my documentation using the `docker run --rm -it -v ${PWD}:/docs mycustomimage` command, my documentation is served on docker host ip address 0.0.0.0:8000.
Is it possible to change this ip address to the actual IP:port address of the machine on which the documentation is served (before the build).
I tried running the container in a custom network, but the ip always defaults to 0.0.0.0:8000.

I don't mean port mapping (I know how to do that), but the change of the actual docker host ip for Material for Mkdocs docker server.

**Comment by squidfunk** [comment, marked as answer]
You should __never use the Docker image for serving the website in production__. It's only meant for building and previewing the site.

**Reply by lukaszgiez** [reply]
I am aware of that. I am using the Docker for serving in pre-roduction stage while we are working on documentation with several people and we have a tool plugged in (Hypothesis) for reviewing content.
The docker server is super handy for quick deployment internally, but the Hypothesis client (plugged in the browser) reads the docker host IP.
Are there any variables we could tweak to change this?

**Reply by squidfunk** [reply]
No, and not planned. You could build your own Docker image if necessary.

**Reply by lukaszgiez** [reply]
Thanks for replying!
I have examined your Dockerfile, and I see the CMD command at the end where you specify the `dev_addr`. I have experimented with various options, from `localhost:8000`, `127.0.0.1:8000` to providing my server address. The image builds fine but none of these work with the documentation once I serve it via this image.
Would you have any tips what I can try out in the Dockerfile so that the page is served on the host IP?
Basically I am trying to avoid the situation in which the page canonical url is set to 0.0.0.0:8000 in the header.

**Reply by lukaszgiez** [reply]
OK, I went ahead and revised our way of working with the documentation. Gitlab builds the output, and then docker nginx serves the html. IP works as expected.
